src/NuGet.Core/NuGet.Commands/RestoreCommand/LockFileBuilderCache.cs

```diff
@@ -76,6 +76,11 @@ public ContentItemCollection GetContentItems(LockFileLibrary library, LocalPacka
         /// </summary>
         internal LockFileTargetLibrary GetLockFileTargetLibrary(RestoreTargetGraph graph, NuGetFramework framework, LocalPackageInfo localPackageInfo, string aliases, LibraryIncludeFlags libraryIncludeFlags, Func<LockFileTargetLibrary> valueFactory)
         {
+            // Comparing RuntimeGraph for equality is very expensive,
+            // so in case of a request where the RuntimeGraph is not empty we avoid using the cache.
+            if (graph.RuntimeIdentifier != null)
```

Comment:
I think we use string.IsNullOrempty in most of our checks.

Reply:
👍